File: parse.py
#!/usr/bin/env python
# encoding: utf-8

import requests, urllib, json

appkey = ""
params = {
   "appkey": appkey,
   "url":"",
}
def get(url):
    params["url"] = url;
    api_url = "https://api.vnil.cn/api/parse/deal?" + urllib.parse.urlencode(params)

    msg = {"code": 0, "msg": "", "body": ""}

    response = requests.get(url=api_url, timeout=30)

    if response.status_code != 200:
	 	msg['code'] = 1
	 	msg["msg"] = "请求出现问题"
	 	return msg
    # result = json.loads(response.text)      如果你直接拿到系统中使用请将返回参数直接转为json
   	result = response.text  # 如果你不需要转换json，则直接接受数据并返回
   	return result


def post(url):
    params["url"] = url
    api_url = "https://api.vnil.cn/api/parse/deal"

    msg = {"code": 0, "msg": "", "body": ""}

    response = requests.post(url=api_url, data=params, timeout=30)
    if response.status_code != 200:
	 	msg['code'] = 1
	  	msg["msg"] = "请求出现问题"
	  	return msg
    # result = json.loads(response.text)      如果你直接拿到系统中使用请将返回参数直接转为json
    result = response.text  # 如果你不需要转换json，则直接接受数据并返回
    return result

print(get("http://xhslink.com/f0aZc"))
#print(post("http://xhslink.com/f0aZc"))

File: parse.php
<?php
// https://www.vnil.cn开发者后台生成的appkey
$appkey = '';
	
//需要解析的url
$url = '';

$param = [
	'appkey'	=> $appkey,
	'url'		=> $url,
];

//得到请求的地址：https://api.vnil.cn/api/parse/deal?appkey=appkey&url=url

$apiUrl = 'https://api.vnil.cn/api/parse/deal?'.http_build_query($param);

$ch = curl_init();
curl_setopt ( $ch, CURLOPT_URL, $apiUrl );
curl_setopt ( $ch, CURLOPT_SSL_VERIFYPEER, FALSE );
curl_setopt ( $ch, CURLOPT_SSL_VERIFYPEER, 0 );
curl_setopt ( $ch, CURLOPT_SSL_VERIFYHOST, 0 );
curl_setopt ( $ch, CURLOPT_MAXREDIRS, 5 );
curl_setopt ( $ch, CURLOPT_RETURNTRANSFER, 1 );
curl_setopt ( $ch, CURLOPT_FOLLOWLOCATION, 1 );
curl_setopt ( $ch, CURLOPT_TIMEOUT, 10 );
$content = curl_exec( $ch );
curl_close ( $ch);

print_r($content);
